lib/spack/spack/test/compilers/basics.py

```diff
@@ -815,3 +815,33 @@ class MockPackage(object):
     pkg = MockPackage()
     with pytest.raises(OSError):
         compiler.setup_custom_environment(pkg, env)
+
+
+@pytest.mark.enable_compiler_verification
+def test_compiler_executable_verification_raises(tmpdir):
+    compiler = MockCompiler()
+    compiler.cc = '/this/path/does/not/exist'
+
+    with pytest.raises(spack.compiler.CompilerAccessError):
+        compiler.verify_executables()
+
+
+@pytest.mark.enable_compiler_verification
+def test_compiler_executable_verification_success(tmpdir):
+    def prepare_executable(name):
+        real = str(tmpdir.join('cc').ensure())
+        fs.set_executable(real)
+        setattr(compiler, name, real)
+
+    # setup mock compiler with real paths
+    compiler = MockCompiler()
+    for name in ('cc', 'cxx', 'f77', 'fc'):
```

Comment:
(comment) the tests you added cover most cases except I think perhaps not the case where only a subset of executables are defined: if a subset of compilers is defined, and all of them are defined properly, then the check should pass. Not essential but it would improve the testing (and it looks like you were careful to check that in the logic).

Reply:
There is a test for that. It's the second half of this test, when I set `compiler.cc` to None